ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -1812,7 +1812,12 @@ private static void processBaseDir(Path baseDir, ValidWriteIdList writeIdList, V
       directory.getAbortedWriteIds().add(parsedBase.writeId);
       return;
     }
-    if (directory.getBase() == null || directory.getBase().getWriteId() < writeId) {
+    if (directory.getBase() == null || directory.getBase().getWriteId() < writeId
+      // If there are two competing versions of a particular write-id, one from the compactor and another from IOW, 
+      // always pick the compactor one once it is committed.
+      || directory.getBase().getWriteId() == writeId && parsedBase.getVisibilityTxnId() > 0 
```

Comment:
Just kind of a nit: there's an `isCompactedBase` method you could use instead of `parsedBase.getVisibilityTxnId() > 0`. It doesn't do much more but it would make this more readable

Reply:
fixed